test/attr/execution_behavior_attrs.swift

```diff
@@ -125,17 +125,70 @@ struct IsolatedType {
   @concurrent func test() async {}
 }
 
-_ = { @concurrent in // Ok
+// `@concurrent` on closures does not contribute to `async` inference.
+do {
+  _ = { @concurrent in }
+  // expected-error@-1:9 {{cannot use @concurrent on non-async closure}}{{none}}
+  _ = { @concurrent () -> Int in }
+  // expected-error@-1:9 {{cannot use @concurrent on non-async closure}}{{none}}
+  _ = { @concurrent () async in }
+  _ = { @concurrent in await awaitMe() }
```

Comment:
Presence of `throws` precludes effect inference from the body, might be good to add a test case where we have `@concurrent` + `throws` without async, to make sure that everything works as expected.